### DISCUSSION TITLE
Concern about lack of "state" and "prompt" in requests.

### DISCUSSION THREAD

**Discussion by chrisdewa** [nwunderly/starlette-discord#5] [Q&A] [answered]
As the title say, i see that the class in  `oauth.py` supports creating states to prevent coss site forgeries but i don't see any indication that `DiscordOAuthClient` actually adds  the state to the url and i see no way to get it on callback to check if it's valid. 

Also, discord documentation says you can add a `prompt` parameter to the url which is also missing.

**Comment by nwunderly** [comment, marked as answer]
Just pushed a fix to here and PyPI, v0.1.1 adds this and fixes #2. You can see an example of how to use it [here](https://github.com/nwunderly/starlette-discord/blob/master/tests/test_state_fastapi.py).

**Reply by chrisdewa** [reply]
Amazing! Thanks
